src/dialogs/more-info/controls/more-info-input_timetable.ts

```diff
@@ -0,0 +1,191 @@
+import "@material/mwc-icon-button/mwc-icon-button";
+import "../../../components/ha-svg-icon";
+import "../../../components/ha-switch";
+import type { HaSwitch } from "../../../components/ha-switch";
+import "../../../components/paper-time-input";
+import { mdiDelete, mdiPlus } from "@mdi/js";
+import { HassEntity } from "home-assistant-js-websocket";
+import { BINARY_STATE_ON, BINARY_STATE_OFF } from "../../../common/const";
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  PropertyValues,
+  TemplateResult,
+} from "lit-element";
+import { HomeAssistant } from "../../../types";
+
+declare type Event = {
+  time: string;
+  state: string;
+};
+
+@customElement("more-info-input_timetable")
+class MoreInfoInputTimetable extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({
+    attribute: false,
+    hasChanged(newVal: HassEntity, oldVal: HassEntity) {
+      let changed = false;
+      if (newVal) {
+        if (!oldVal) {
+          changed = true;
+        } else {
+          const new_timetable = newVal.attributes.timetable;
+          const old_timetable = oldVal.attributes.timetable;
+          if (JSON.stringify(old_timetable) !== JSON.stringify(new_timetable)) {
+            changed = true;
+          }
+        }
+      }
+      return changed;
+    },
+  })
+  public stateObj?: HassEntity;
+
+  private timetable: Array<Event> = [];
+
+  private disable_reconfig = false;
+
+  protected render(): TemplateResult {
+    if (!this.hass || !this.stateObj) {
+      return html``;
+    }
+    this.disable_reconfig = true;
+    return html`
+      ${this.timetable.map((event: Event, index: number) =>
+        this._row(event, index)
+      )}
+      <div class="add">
+        <mwc-icon-button
+          @click=${() => {
+            this._add();
+          }}
+        >
+          <ha-svg-icon .path=${mdiPlus}></ha-svg-icon>
+        </mwc-icon-button>
+        <p>
+          ${this.hass!.localize(
+            "ui.dialogs.more_info_control.input_timetable.add"
+          )}
+        </p>
+      </div>
+    `;
+  }
+
+  updated() {
+    this.disable_reconfig = false;
+  }
+
+  private _row(event: Event, index: number) {
```

Comment:
I would rename to `_renderRow()` so it's clearer what it does.

Reply:
Done